### PR TITLE
Regression integration

### PR DESCRIPTION
**Pull request by samuel-barrett**

### CODE REVIEW

**Comment by Erik-McKelvey** [comment]
Add 
`matplotlib==3.7.2`
to requirements.txt pls

**Comment by sispuella** [comment]
Please check conflict. 
Also grouping by some unique column in feature engineering will result in all new features being empty.

File: enrollment_predictions/models/auto_regressor_dec_tree.py [hunk 1]
```diff
@@ -1,180 +1,310 @@
 import pandas as pd
 import numpy as np
+import matplotlib.pyplot as plt
```
Comment (Erik-McKelvey) [inline enrollment_predictions/models/auto_regressor_dec_tree.py]: Could you move this import to the 
```
if __name__ == "__main__":
    import matplotlib.patches as mpatches
```

File: enrollment_predictions/models/auto_regressor_dec_tree.py [hunk 2]
```diff
@@ -1,180 +1,309 @@
 import pandas as pd
 import numpy as np
+from sklearn.metrics import mean_absolute_error, mean_squared_error
 from sklearn.impute import SimpleImputer
+from lightgbm import LGBMRegressor
```
Comment (Erik-McKelvey) [inline enrollment_predictions/models/auto_regressor_dec_tree.py]: Could you remove this line please

Comment (Erik-McKelvey) [reply enrollment_predictions/models/auto_regressor_dec_tree.py]: Or add the requirement to `requirements.txt` if it will be used

**Review by Erik-McKelvey — APPROVED** [review]

**Pull request merged**